### PR TITLE
Clean up, fix up and refactor IO streams and channels

### PR DESCRIPTION
**Pull request by medb**

### CODE REVIEW

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java [hunk 1]
```diff
@@ -17,202 +17,156 @@
...
-          byte b = singleReadBuf[0];
-          totalBytesRead++;
-          statistics.incrementBytesRead(1);
-          statistics.incrementReadOps(1);
-          result = (b & 0xff);
+      int numRead = channel.read(ByteBuffer.wrap(singleReadBuf));
+      if (numRead == -1) {
+        response = -1;
+      } else {
+        if (numRead != 1) {
```
Comment (mprashanthsagar) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:98]: Can we collapse with ` else if ` instead

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java [hunk 2]
```diff
@@ -17,202 +17,156 @@
...
   public synchronized void seek(long pos) throws IOException {
+    logger.atFiner().log("seek(%d)", pos);
     long curPos = getPos();
     long diff = pos - curPos;
     if (diff > 0) {
       streamStatistics.seekForwards(diff);
     } else {
       streamStatistics.seekBackwards(diff);
     }
-    synchronized (this) {
```
Comment (mprashanthsagar) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:159]: Any particular reason we are removing the synchronized block ( I am unaware of why it was there in the first place )

Comment (medb) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java:159]: This block is already in `synchronized` method, so this block is redundant.

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java (hunk 1 above)
Comment (medb) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java]: Done.

**Review by mprashanthsagar — APPROVED** [review]

**Pull request merged**